.circleci/template.yml

```diff
@@ -0,0 +1,884 @@
+version: 2.1
+
+references:
+  - &OTP24 cimg/elixir:1.12.3  # Contains Erlang 24.0.5
+  - &OTP23 cimg/elixir:1.11.4  # Contains Erlang 23.1
+  - &ENTRYPOINT ["/bin/sh", "-c", "eval ${INSTALL_DEPS_CMD:-echo} && echo __INJECT_FILES__ | eval ${BASE32DEC:-base32 --decode} | bash"]
+  # Caches created via the save_cache step are stored for up to 15 days
+  - &CERT_KEY certs-cache-{{ checksum "tools/ssl/Makefile" }}-{{ checksum "cert_date" }}-v5
+  - &DEPS_CACHE_KEY deps-cache-{{ checksum "rebar.lock" }}-{{ checksum "big_tests/rebar.lock" }}-{{ checksum "otp_version" }}-v4
+  - &BUILD_CACHE_KEY build-cache-{{ .Branch }}-{{ .Revision }}-{{ checksum "otp_version" }}-v5
+
+# list of references to be used inside executors block
+containers:
+  - &mysql_container
+    image: mysql:8.0.20
```

Comment:
Can we make sure this uses the same version that we'll use locally and also in github actions?
When creating the config in the setup step, we can expand some vars from here https://github.com/esl/MongooseIM/blob/master/tools/db-versions.sh, likewise for the containers below.

I think this one would be the most important comment I'd say is necessary for this PR, most other details we can do in the future in fine-grained separate PRs 😁